numpy/_core/src/common/npy_atomic.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Provides wrappers around C11 standard library atomics and MSVC intrinsics
+ * to provide basic atomic load and store functionality. This is based on
+ * code in CPython's pyatomic.h, pyatomic_std.h, and pyatomic_msc.h
+ */
+
+#ifndef NUMPY_CORE_SRC_COMMON_NPY_ATOMIC_H_
+#define NUMPY_CORE_SRC_COMMON_NPY_ATOMIC_H_
+
+#include "numpy/npy_common.h"
+
+#if __STDC_VERSION__ >= 201112L && !defined(__STDC_NO_ATOMICS__)
+// TODO: support C++ atomics as well if this header is ever needed in C++
```

Comment:
The code to do this is in the CPython header this is cribbed from. It would be dead code if I included it.